From: Corey Wagehoft <corey.wagehoft@gmail.com>
Date: Tue, 7 Jul 2026 00:00:00 -0500
Subject: [PATCH] dt-overlays: raven: add overlay fragment

Rename GPIO lines for the RAVEN CM4 carrier board (LC76G GPS on
UART0, 1PPS input on GPIO18, GPS jamming indicator on GPIO4).

---
 .../arm/boot/dts/overlays/raven-overlay.dts | 32 ++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/raven-overlay.dts

--- /dev/null
+++ b/arch/arm/boot/dts/overlays/raven-overlay.dts
@@ -0,0 +1,32 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2711";
+
+	fragment@0 {
+		target = <&gpio>;
+		__overlay__ {
+			gpio-line-names =
+				"ID_SDA", "ID_SCL", "SDA1", "SCL1",
+				"GPS_JAM_IND",            /* GPIO4  - LC76G jamming indicator */
+				"GPIO5", "GPIO6", "SPI_CE1_N",
+				"SPI_CE0_N", "SPI_MISO", "SPI_MOSI", "SPI_SCLK",
+				"GPIO12", "GPIO13",
+				"GPS_UART_TX",            /* GPIO14 - CM4 TXD0 -> LC76G RX */
+				"GPS_UART_RX",            /* GPIO15 - CM4 RXD0 <- LC76G TX */
+				"GPIO16", "GPIO17",
+				"GPS_PPS",                /* GPIO18 - 1PPS in */
+				"GPIO19", "GPIO20", "GPIO21", "GPIO22", "GPIO23",
+				"GPIO24", "GPIO25", "GPIO26", "GPIO27",
+				"RGMII_MDIO", "RGMII_MDC", "CTS0", "RTS0",
+				"TXD0", "RXD0", "SD1_CLK", "SD1_CMD",
+				"SD1_DATA0", "SD1_DATA1", "SD1_DATA2", "SD1_DATA3",
+				"PWM0_MISO", "PWM1_MOSI", "STATUS_LED_G_CLK", "SPIFLASH_CE_N",
+				"SDA0", "SCL0",
+				"RGMII_RXCLK", "RGMII_RXCTL", "RGMII_RXD0", "RGMII_RXD1",
+				"RGMII_RXD2", "RGMII_RXD3", "RGMII_TXCLK", "RGMII_TXCTL",
+				"RGMII_TXD0", "RGMII_TXD1", "RGMII_TXD2", "RGMII_TXD3";
+		};
+	};
+};
